pkg/preparation/storacha/storacha.go

```diff
@@ -77,10 +95,27 @@ func (a API) SpaceBlobAddShardsForUpload(ctx context.Context, uploadID id.Upload
 			locationCommitment,
 		)
 		if err != nil {
-			return fmt.Errorf("failed to replicate shard %s: %w", shard.CID(), err)
+			return fmt.Errorf("failed to replicate shard %s: %w", shard.ID(), err)
+		}
+
+		pieceDigest, _, err := commpCalc.Digest()
+		if err != nil && !strings.Contains(err.Error(), "insufficient state accumulated") {
```

Comment:
Can we just use `shard.Size()` to detect that it's too small and not send it to `commpCalc`?

Reply:
We can. I didn't want to depend on the magic number, but this way we're depending on a magic error string, so maybe the magic number and not calling code that won't run is better.

Reply:
[Oh, never mind, it's even exported!](https://github.com/storacha/guppy/pull/136#discussion_r2411674601)